### ISSUE TITLE
Data is leaked when logout and login with different user

### ISSUE THREAD

**Issue by rogup** [DigitalCommons/land-explorer-front-end#228]
**Describe the bug**
We don't clear the app state when logging out e.g. the current map. So if we log in with a different user, some of that (potentially sensitive) data is visible to the new user, and they can even edit some of it.

**To Reproduce**
1. Create a map with a marker and save it
2. Logout and login with a different user
3. The map owned by the first user is still visible.
4. See saving error, since the map cannot be saved as the second user. However, the marker name/description can be edited.
5. Log back in to the first user and see that the marker was indeed successfully edited!

**Expected behaviour**
All user data should be wiped from the app's Redux state when logging out, to avoid this happening.

Also, we should lock down "/api/user/edit/marker" and other similar APIs in the backend so that only user with access can control them, since they are currently susceptible to anyone editing data if they guess the object UUID.

**Comment by ms0ur1s** [comment]
Definitely took a bit longer on this than expected (add one pinch of misunderstanding and a sprinkle of redux learning curve), but with a good resolution.

The forced refresh on logout suggested as last resort by @King-Mob worked fine but it looked hella janky, not a good look.

Next I tried setting the user data to its initial state (via the UserReducer) ` case "LOG_OUT":  return { ...INITIAL_STATE, }; `  which cleared the user data but not the map data. So I followed the same pattern with the MapReducer which had the unforeseen result of deleting the last map open before logging out. Something I didn't think was possible without using a CRUD operation. This sent me down the fruitless route creating a map action to prevent the map deletion while still setting the map data to its initial state.

```
// Added to save map to backend when user offloads map  
export const offloadMaps = () => {
return async (dispatch, getState) => {
        const currentMapId = getState().mapMeta.currentMapId;
        if (currentMapId) {
            await dispatch(saveCurrentMap());
            dispatch({ type: 'CLEAR_MAP_DATA' });
        }
    }
}
```

It wasn't until digging a little further into Redux and abandoning this line of investigation, that I realised that I could clear all data from the Redux store on logout (probably what @rogup meant rather than me going all around the houses) via the RootReducer. A solution which I found below and works like a charm.

[https://www.digitalocean.com/community/tutorials/redux-reset-state-redux](https://www.digitalocean.com/community/tutorials/redux-reset-state-redux)
